celery/backends/base.py

```diff
@@ -256,6 +256,19 @@ def fail_from_current_stack(self, task_id, exc=None):
             self.mark_as_failure(task_id, exc, exception_info.traceback)
             return exception_info
         finally:
+            if sys.version_info >= (3, 5, 0):
+                while tb is not None:
+                    try:
+                        tb.tb_frame.clear()
```

Comment:
Sorry I'm a little "late to the party" here, but if I may, I want to forward a couple of question for @heedong-jung (perhaps for a future improvement):

1. Could we use [`traceback.clear_frames()`][1], introduced in python 3.4, to avoid manually iterating over and clearing the frame data above?

2. Are we not able to add an integration test, to more explicitly check for memory leaks over task execution?

3. Would it be possible to reuse the code in `traceback_clear`, eeither by allowing passing a traceback object, or perhaps by splitting into separate "extract traceback from exception" and "clear traceback frames"?

4. Do you believe it's related to (or resolves) issue #6058 ?

I believe that with the above information, I can start working on improvements where needed.

[1]: https://docs.python.org/3.5/library/traceback.html#traceback.clear_frames

Reply:
you can come with a PR with appropriate tests

Reply:
clear_frames() is not suitable for solving this problem.

It is related to the memory leak you mentioned.